src/components/annotate/sidePanel/AnnotationForm.tsx

```diff
@@ -53,30 +54,45 @@ const AnnotationForm = (props: Props) => {
     const [initialValues, setInitialValues] = useState<Dict>({});
     const [classFormFields, setClassFormFields] = useState<Dict>({});
 
-    /* OnLoad or change of Annotate Target: Prepare initial form values */
-    useEffect(() => {
+    /* Function to determine which field or class source to use */
+    const DetermineFieldOrClass = () => {
         let targetField: string = '';
         let targetClass: string = '';
-        let motivation: string = '';
-        let classProperties = {};
+        let motivation: string | undefined = 'oa:commenting';
 
-        /* Determine presence of target field or class */
         if (!isEmpty(editAnnotation)) {
-            targetField = editAnnotation['oa:target']['oa:selector']?.['ods:field'] as string;
+            if (editAnnotation['oa:target']['oa:selector']?.['ods:field']) {
+                targetField = editAnnotation['oa:target']?.['oa:selector']?.['ods:field'] as string;
+            } else if (editAnnotation['oa:target']['oa:selector']?.['oa:class']) {
+                targetClass = editAnnotation['oa:target']?.['oa:selector']?.['oa:class'] as string;
+            }
+
             motivation = editAnnotation['oa:motivation'] as string;
         } else if (annotateTarget.targetProperty.type === 'field') {
             targetField = annotateTarget.targetProperty.name;
+            motivation = 'ods:adding';
```

Comment:
It doesn't necessarily be an `ods:adding` annotation right? There could also be a comment annotation on a field or class.

Reply:
Correct, this predefined motivation makes sure at least one is selected, otherwise the annotation form will not be shown (checks if a motivation is present), the user can change this any other motivation by using the dropdown (except for cases where adding or editing is enforced).